daffodil-cli/src/it/resources/org/apache/daffodil/CLI/output/output9.txt

```diff
@@ -1,4 +1,4 @@
-<base14:rabbitHole xmlns:b14="http://b14.com" xmlns:a14="http://a14.com" xmlns:base14="http://baseSchema.com">
+<base14:rabbitHole xmlns:a14="http://a14.com" xmlns:b14="http://b14.com" xmlns:base14="http://baseSchema.com">
```

Comment:
Why does this matter? Order shouldn't matter for these.

Reply:
Most of our tests either use the TDML runner and so attributes are ignored, or they aren't complex enough to require multiple prefix mappings.

This file is the expected output for one of the few tests where we expect the exact output from the Daffodil CLI. Since we've changed the orer of the namespaces to be consistent and repeatable, regardless of Scala version, this test was affected. So order here doesn't matter, but our test is looking for a specific order because it's looking for exact output.

Reply:
Gotcha thanks. A comment to that effect with the test would be helpful.

Reply:
As to you other question, I'm not really sure why the serialization changes causes namespace mapping ordering changes. I suspect copying the list caused some things to be evaluated in a slightly different order and because we used Set/Maps, however things got lazily evaluated changed the order. It's not entirely clear to me, and I couldn't track down the underlying change. But 2.12.13 did backport some Map changes from 2.13.x, so a change in ordering isn't too surprising.

Reply:
So based on this test being order specific, any given new scala library might change the order we got from Sets & Hash Maps, so could have broken this. Copying the map with identity could traverse it and the re-hash might be different because the order of arrival is different. 

So it's very plausible that the serialization change or workarounds changed this.

Reply:
Yep, very likely what happened.

I can't add a comment to the output file without breaking the test since that file is exactly the output of the CLI, but I've added a comment to the test that uses this file.